store/store.go

```diff
@@ -513,6 +552,23 @@ func LoadBlockStoreState(db dbm.DB) cmtstore.BlockStoreState {
 	return bsj
 }
 
+// LoadTxStatus retrieves the status of a transaction from the block store given its hash.
+func (bs *BlockStore) LoadTxStatus(txHash []byte) *cmtstore.TxStatus {
+	bz, err := bs.db.Get(calcTxHashKey(txHash))
+	if err != nil {
+		panic(err)
```

Comment:
Is panic essential at this point? I think it is perceivable that someone request tx status for a wrong/unavailable hash.

Reply:
i think for wrong/unavailable hash we'd get nil

Reply:
> i think for wrong/unavailable hash we'd get nil

Correct, I am proposing the removal of the panic behavior. Alternatively, could you explain why retaining the panic option is beneficial?

Reply:
usually, an error returned from the db is a severe issue, like disk corruption. blockstore generally panics if a read/write operation on the database results in an error, and I'm trying to follow that pattern.

If we want to change this behaviour and return an error instead, we can discuss a potential refactor of the other getters as well. However, i'd say that's beyond the scope of this pr.

Reply:
I understand now. Since the existence of the hash in the previous blocks is already verified in another part of the code, this function will never be invoked with a non-existing hash. All good then!